frontend/packages/console-plugin-sdk/README.md

```diff
@@ -0,0 +1,234 @@
+    properties: {
+      id: 'persistent-storage',
+      title: 'Persistent Storage',
+    },
+  },
+  ...
+];
+
+export default plugin;
+```
+
+See the [demo plugin](frontend/packages/console-demo-plugin) for available Plugin extension points.
+
+For better type checking and code completion, use a type parameter that represents the union of all the extension types consumed by the plugin.
+
+``` es6
+// don't do this
+const plugin: Plugin<any> = [ /* stuff */ ];
+// do this
+const plugin: Plugin<FooExtension | BarExtension> = [ /* stuff */ ];
+// or better yet, this
+type ConsumedExtensions = FooExtension | BarExtension;
+const plugin: Plugin<ConsumedExtensions> = [ /* stuff */ ];
+```
+
+## Monorepo Architecture
+
+Console plugins are 1st class citizens of Console monorepo and therefore reuse the same testing (Jest, Protractor, etc.) infrastructure and conventions.
+See Tests and OWNERS sections below for details.
+
+## Unit Tests
+
+The jest framework is used for unit tests. It will automatically run any file located in a directory named `.../__tests__` or whose name ends with a `.spec.(ts|tsx|js|jsx)` suffix.
+
+``` bash
+  .
+  ├── my-plugin
+  |   ├── src
+  |   |   ├── __tests__
+  |   |   |   ├── some-test.spec.ts
+  |   |   |   └── ...
+  |   |   └── ...
+  |   └── ...
+  └── ...
+```
+
+## Integration Tests
+
+Integration tests use [protractor](http://www.protractortest.org) and must be defined in the `consolePlugin` map, in the plugins `package.json`.
+Test path is relative the plugin root directory. It is possible to include common tests defined in `@console/internal-integration-tests`,
+and use common testing methods defined in the internal package by requiring it in the test.
+
+``` json
+// File: package.json
+
+{
+  ...
+  "consolePlugin": {
+    ...
+    "integrationTests": {
+      "demo": [
+        "@console/internal-integration-tests/tests/base.scenario.ts",
+        "integration-tests/tests/demo.scenario.ts"
+      ]
+    }
+  }
+}
+```
+
+``` bash
+  .
+  ├── my-plugin
+  |   ├── integration-tests
+  |   |   ├── tests
+  |   |   |   ├── demo.scenario.ts
+  |   |   |   └── ...
+  |   |   └── ...
+  |   └── ...
+  └── ...
+```
+
+## OWNERS File
+
+The OWNERS file define a list of maintainers, community plugins must define a list of `reviewers` and `approvers` who will
+be responsible of reviewing and approving pull requests for this plugin. For more in depth information see [prow](https://github.com/kubernetes/test-infra/tree/master/prow) documentation.
+
+The OWNERS file should also include `labels`:
+
+``` yaml
+labels:
+ - component/<plugin-name>
+```
+
+To add a new component labels, raise a PR here: https://github.com/openshift/release/blob/master/cluster/ci/config/prow/labels.yaml
+
+## Common vs. Community Plugins
+
+Plugins can be owned by console or community teams, plugins maintained by the console team will not have an
+`OWNERS` file, and by convention use a directory named `console-<some-name>`. 
```

Comment:
"Plugins can be owned by console or community teams, plugins maintained by the console team will not have an `OWNERS` file, and by convention use a directory named `console-<some-name>`."

"Plugins can be owned either by the Console team or community teams. Plugins maintained by the Console team will not have an `OWNERS` file and will use a directory named `console-<some-name>` by convention."